pkg/splunk/reconcile/service.go

```diff
@@ -22,21 +22,33 @@ import (
 )
 
 // ApplyService creates or updates a Kubernetes Service
-func ApplyService(client ControllerClient, service *corev1.Service) error {
+func ApplyService(client ControllerClient, revised *corev1.Service) error {
 	scopedLog := log.WithName("ApplyService").WithValues(
-		"name", service.GetObjectMeta().GetName(),
-		"namespace", service.GetObjectMeta().GetNamespace())
+		"name", revised.GetObjectMeta().GetName(),
+		"namespace", revised.GetObjectMeta().GetNamespace())
 
-	namespacedName := types.NamespacedName{Namespace: service.GetNamespace(), Name: service.GetName()}
+	namespacedName := types.NamespacedName{Namespace: revised.GetNamespace(), Name: revised.GetName()}
 	var current corev1.Service
 
 	err := client.Get(context.TODO(), namespacedName, &current)
-	if err == nil {
-		// found existing Service: do nothing
-		scopedLog.Info("Found existing Service")
-	} else {
-		err = CreateResource(client, service)
+	if err != nil {
+		return CreateResource(client, revised)
 	}
 
-	return err
+	// found existing Service: do nothing
+	scopedLog.Info("Found existing Service")
```

Comment:
this comment could probably just be removed..

I don't recall why it wasn't applying updates for services. i imagine partly because we didn't expect them to change, but also when i tried flipping it, it was *always* making updates to services, even when there were no differences. looks like you tackled this below with MergeServiceSpecUpdates, but just wanted to make sure you tested that it's not updating services every time it reconciles.

Reply:
Will remove the comment. I tested it wasn't making unnecessary updates. I had to initialize the targetPort field since its optional.